linux/debian/9/foxpass_setup.py

```diff
@@ -61,12 +60,12 @@ def main():
 
     apt_get_update()
     install_dependencies()
-    write_foxpass_ssh_keys_script(apis, args.api_key)
-    write_nslcd_conf(uris, args.base_dn, binddn, args.bind_pw, args.ldap_connections, args.idle_timelimit)
-    augment_sshd_config()
-    augment_pam()
-    fix_nsswitch()
-    fix_sudo(args.sudoers_group, args.require_sudoers_pw, args.update_sudoers)
+    write_foxpass_ssh_keys_script(apis, args.api_key, args.debug)
```

Comment:
Instead of passing `debug` flag to all the functions, do you think it makes sense to check for the flag before calling the function and after calling the function for any differences to log. The debug in that case can just be handled here and needn't be in the individual function.

```
if debug:
    old = record_old_file('')
    write_foxpass_ssh_keys_script(apis, args.api_key)
    new = record_new_file('')
    record_diff(old, new)
```